Formula/unisonlang.rb

```diff
@@ -0,0 +1,49 @@
+require "language/node"
+
+class Unisonlang < Formula
+  desc "Friendly programming language from the future"
+  homepage "https://unison-lang.org/"
+  url "https://github.com/unisonweb/unison.git",
+      tag:      "release/M3",
+      revision: "97cce8d23147df30b0f80ae745968db09f4e1a44"
+  version "1.0.0-M3"
+  license "MIT"
+  head "https://github.com/unisonweb/unison.git", branch: "trunk"
+
+  depends_on "haskell-stack" => :build
+  depends_on "node" => :build
+
+  on_linux do
+    depends_on "xz" => :build
+  end
+
+  resource "unison-ui" do
+    url "https://github.com/unisonweb/codebase-ui.git",
+        tag:      "release/M3",
+        revision: "05df3a235dec7d4a03ff1c588a32eb6daca288d4"
+  end
+
+  def install
+    # Build the ucm binary
+    system "stack", "build", "--flag", "unison-parser-typechecker:optimized"
+
+    # Install the ucm binary
+    local_install_root = Pathname.new `stack path | awk '/local-install-root/{print $2}'`.to_s.strip
```

Comment:
Please don't use backticks but use the safe_read functions.

Reply:
Not sure I understand how `safe_read` would work here? It's not reading file, it's determining the directory the build output will live in, that we should copy it from

Reply:
I think he means `safe_popen_read`, but since your command is a pipeline, maybe something like this (WARNING: Ruby n00b):
```
local_install_root = Pathname.new Utils.safe_popen_read("/bin/sh", "-c", "#{bin}/stack path | awk '/local-install-root/{print $2}').strip
```